app/models/compute_resources/foreman/model/ovirt.rb

```diff
@@ -6,8 +6,7 @@ class Ovirt < ComputeResource
     validates :url, :format => { :with => URI::DEFAULT_PARSER.make_regexp }, :presence => true,
               :url_schema => ['http', 'https']
     validates :user, :password, :presence => true
-    before_create :update_public_key
-    after_validation :update_available_operating_systems unless Rails.env.test?
+    after_validation :connect, :update_available_operating_systems unless Rails.env.test?
```

Comment:
You have changed the behavior for test environment: `update_public_key` was called even for test environment, but now `connect` will not be called.
Is it intentional?

Reply:
No but I wonder if we should actually call `update_public_key` for test environment.
I could add it just for test environment. I can't use `connect` in test environment because there isn't a real ovirt to connect to.

Reply:
If we are sure we can't use real ovirt for tests, then I think it's OK to leave it here.

Reply:
we can't